code/obj/item/clothing/gloves.dm

```diff
@@ -234,6 +234,16 @@ ABSTRACT_TYPE(/obj/item/clothing/gloves)
 		..()
 		setProperty("heatprot", 7)
 
+/obj/item/clothing/gloves/black/attackby(obj/item/W as obj, mob/user as mob)
+	if (istool(W, TOOL_CUTTING | TOOL_SNIPPING))
+		user.visible_message("<span class='notice'>[user] cuts off the fingertips from the [src].</span>")
+		if(src.loc == user)
+			user.u_equip(src)
+		src.set_loc(user)
+		var/obj/item/clothing/gloves/black/I = new /obj/item/clothing/gloves/fingerless
+		user.put_in_hand_or_drop(I)
+		qdel(src)
```

Comment:
* your type casting this object to black gloves needlessly. That said, you don't actually even need that var
* setting the black gloves `loc` to the user contents is also unneeded as `qdel(src)` already warps the atom to null space

```suggestion
		qdel(src)
		user.put_in_hand_or_drop(new /obj/item/clothing/gloves/fingerless)
```

Reply:
is the double qdel on purpose?

Reply:
I'm unsure where there is a second. I removed one and moved it up.